### PR TITLE
Clean up CMake configs

### PR DESCRIPTION
**Pull request by LukasWoodtli**
Some clean-ups. Also add sanitized integration tests.

### CODE REVIEW

File: wakaama.cmake [hunk 1]
```diff
@@ -5,17 +5,9 @@ set(WAKAAMA_EXAMPLE_DIRECTORY "${WAKAAMA_TOP_LEVEL_DIRECTORY}/examples")
...
-
-if(NOT WAKAAMA_MODE_SERVER
-   AND NOT WAKAAMA_MODE_BOOTSTRAP_SERVER
-   AND NOT WAKAAMA_MODE_CLIENT
-)
-    message(WARNING "No mode selected. Defaulting to 'WAKAAMA_MODE_SERVER'")
-    set(WAKAAMA_MODE_SERVER ON)
-endif()
+option(WAKAAMA_MODE_SERVER "Enable LWM2M Server interfaces" ON)
+option(WAKAAMA_MODE_BOOTSTRAP_SERVER "Enable LWM2M Bootstrap Server interfaces" ON)
```
Comment (rettichschnidi) [inline wakaama.cmake:9]:
```suggestion
option(WAKAAMA_MODE_BOOTSTRAP_SERVER "Enable LwM2M Bootstrap Server interfaces" ON)
```

File: wakaama.cmake [hunk 2]
```diff
@@ -5,17 +5,9 @@ set(WAKAAMA_EXAMPLE_DIRECTORY "${WAKAAMA_TOP_LEVEL_DIRECTORY}/examples")
...
+option(WAKAAMA_MODE_CLIENT "Enable LWM2M Client interfaces" ON)
```
Comment (rettichschnidi) [inline wakaama.cmake:10]:
```suggestion
option(WAKAAMA_MODE_CLIENT "Enable LwM2M Client interfaces" ON)
```

**Review by rettichschnidi — CHANGES_REQUESTED** [review]

**Review by rettichschnidi — APPROVED** [review]
LGTM

File: .github/workflows/build_and_test.yaml [hunk 3]
```diff
@@ -18,6 +18,7 @@ jobs:
       matrix:
         sanitizer: ["address", "leak", "thread", "undefined"]
         compiler: ["clang-18", "gcc-13"]
+        example: ["bootstrap_server", "client/udp", "client/tinydtls", "client/raw_block1", "lightclient", "server"]
```
[the remark this answers is not in the document]
Comment (LukasWoodtli) [reply .github/workflows/build_and_test.yaml:21]: Good to know! Thanks!

**Pull request merged**